aiopg/utils.py

```diff
@@ -137,27 +137,41 @@ class ClosableQueue:
     are consumed.
     """
 
-    def __init__(self, queue: asyncio.Queue):
+    __slots__ = ("_loop", "_queue", "_close_event")
+
+    def __init__(self, queue: asyncio.Queue, loop: asyncio.AbstractEventLoop):
+        self._loop = loop
         self._queue = queue
-        self._close_exception = asyncio.Future()  # type: asyncio.Future[None]
+        self._close_event = loop.create_future()
 
     def close(self, exception: Exception) -> None:
-        if not self._close_exception.done():
-            self._close_exception.set_exception(exception)
+        if self._close_event.done():
+            return
+        self._close_event.set_exception(exception)
 
     async def get(self) -> Any:
-        loop = get_running_loop()
-        get = loop.create_task(self._queue.get())
+        if self._close_event.done():
+            try:
+                return self._queue.get_nowait()
```

Comment:
So, we extracting only one event. And what if there are many? Maybe skip this totally?